### PR TITLE
add `sciarg` dataset

### PR DESCRIPTION
**Pull request by ArneBinder**
In this PR, we add SciArg - an argumentation-mining dataset, according to the description in #10.

Note: This requires `pie-dataset>=0.4.0` (in detail: #57, #58, #59, #60).

TODO:
 - [x] complete tests
 - [x] complete PIE dataset card
 - [x] edit HF dataset card, if needed: see https://huggingface.co/datasets/DFKI-SLT/brat/commit/bb8c37d84ddf2da1e691d226c55fef48fd8149b5
 - [x] ~maybe re-think~ adhust how `parts_of_same` relations and dataset variant `name=merge_fragmented_spans` play together @ArneBinder
   - [x] overwrite configs with a single `default` one, but with parameter `merge_fragmented_spans=true` (and set `DOCUMENT_CONVERTERS` instead of overwriting `document_converters`)
   - ~[ ] add entry for `...With LabeledMultiSpan...` to `DOCUMENT_CONVERTERS`~ will be added in a follow-up PR

### CODE REVIEW

File: dataset_builders/hf/sciarg/README.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+rhetorical dimensions of scientific writing." ([Lauscher et al., 2018](<(https://aclanthology.org/W18-5206/)>), pp. 40-41)
+
+### Supported Tasks and Leaderboards
+
+- **Tasks**: Argumentation Mining, Component Identification, Relation Identification
+- **Leaderboard:** [More Information Needed](https://github.com/huggingface/datasets/blob/master/CONTRIBUTING.md#how-to-contribute-to-the-dataset-cards)
+
+### Languages
+
+The language in the dataset is English (scientific academic publications on computer graphics).
```
Comment (ArneBinder) [inline dataset_builders/hf/sciarg/README.md:21]: The sections `Dataset Description`, `Dataset Summary`, `Supported Tasks and Leaderboards`, and `Languages should` go the PIE dateset card.

File: dataset_builders/hf/sciarg/README.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+  - `text`: the text that belongs to an ADU, a `string` feature
+  - `type`: the label of the ADU, a `string` feature (See [ADU type list](#components))
+  - `id`: the assigned id to the ADU, a `string` feature
+- `relations`: binary relation annotations that mark the argumentative relations that hold between a head and a tail ADU, a `list` of `dictionay` feature. Each entry has the following fields:
+  - `id`: the assigned id to a relation, a `string` feature
+  - `head`: the first element in a relation, consists of:
+    - `ref_id`: the id of the ADU
+    - `role`: the ordinal id of the unit in the argument
+  - `tail`: the second element in a relation, as `head`, consists of the fields: `ref_id` and `role`.
+  - `type`: the label of the relation, a `string` feature (See [relation type list](#relations))
```
Comment (ArneBinder) [inline dataset_builders/hf/sciarg/README.md:73]: The whole `Dataset Instances` should go to https://huggingface.co/datasets/DFKI-SLT/brat (that I meant with "Update the HF dataset card, if necessary" in the PR description TODOs). But `Data Fields` needs a bit of adjustment (any occurence of `ADU` needs to be sth like "span annotation").

File: dataset_builders/hf/sciarg/README.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+### Data Splits
```
Comment (ArneBinder) [inline dataset_builders/hf/sciarg/README.md:75]: The remaining content of this file needs to be integrated into to the PIE dataset card.

File: dataset_builders/pie/sciarg/README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# PIE Dataset Card for "sciarg"
+
+This is a [PyTorch-IE](https://github.com/ChristophAlt/pytorch-ie) wrapper for the SciArg dataset.
+
+More information about the dataset can be found in the [HF dataset card](https://huggingface.co/datasets/DFKI-SLT/sciarg/blob/main/README.md). However, this is not the wrapper for the [SciArg Huggingface dataset loading script](https://huggingface.co/datasets/DFKI-SLT/sciarg), but rather [BRAT Huggingface dataset loading script](https://huggingface.co/datasets/DFKI-SLT/brat).
+
+Therefore, the current SciArg dataset follows the data structure described in [PIE-Brat dataset card](https://huggingface.co/datasets/pie/brat).
```
Comment (ArneBinder) [inline dataset_builders/pie/sciarg/README.md:7]: lower case, when we speak about the PIE (or HF) dataset
```suggestion
Therefore, the current `sciarg` dataset follows the data structure described in [PIE-Brat dataset card](https://huggingface.co/datasets/pie/brat).
```

File: dataset_builders/pie/sciarg/README.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+- `metadata` (dictionary, optional)
+
+and the following annotation layers:
+
+- `spans` (annotation type: `LabeledMultiSpan` in the case of `BratDocument` and `LabeledSpan` and in the case of `BratDocumentWithMergedSpans`, target: `text`)
+- `relations` (annotation type: `BinaryRelation`, target: `spans`)
+- `span_attributes` (annotation type: `Attribute`, target: `spans`)
+- `relation_attributes` (annotation type: `Attribute`, target: `relations`)
+
+See [here](https://github.com/ChristophAlt/pytorch-ie/blob/main/src/pytorch_ie/annotations.py) for the common annotation type definitions, and [here](https://huggingface.co/datasets/pie/brat/blob/main/README.md?code=true#L30-L41) for the special-type definitions of `LabeledMultiSpan` and `Attribute`.
```
Comment (ArneBinder) [inline dataset_builders/pie/sciarg/README.md:33]: for `Dataset Variants` and `Data Schema`, just reference the respective sections in the PIE `brat` dataset card

**Review by ArneBinder — COMMENTED** [review]
The dataset cards got a bit mixed up, see below. In short: we do not have a HF `sciarg` dataset, so we can not have a HF dataset card for it.

File: dataset_builders/pie/sciarg/README.md (hunk 5 above)
Comment (idalr) [reply dataset_builders/pie/sciarg/README.md]: edit in [22be467]

File: dataset_builders/hf/sciarg/README.md (hunk 3 above)
Comment (idalr) [reply dataset_builders/hf/sciarg/README.md]: moved from hf/readme to pie/readme in 22be467

File: dataset_builders/hf/sciarg/README.md (hunk 1 above)
Comment (idalr) [reply dataset_builders/hf/sciarg/README.md]: moved from hf/readme to pie/readme in 22be467

File: dataset_builders/hf/sciarg/README.md (hunk 2 above)
Comment (idalr) [reply dataset_builders/hf/sciarg/README.md]: - imported the current BRAT HF dataset card in d722685
- added contents in the section `Data Instances` and `Data Fields` in b5b580d

**Pull request merged**